src/main/java/org/jabref/gui/maintable/SmartConstrainedResizePolicy.java

```diff
@@ -22,75 +23,158 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SmartConstrainedResizePolicy.class);
 
+    private Map<TableColumnBase, Double> expansionShare = new HashMap<>();
+
     @Override
     public Boolean call(TableView.ResizeFeatures prop) {
         if (prop.getColumn() == null) {
-            return initColumnSize(prop.getTable());
+            // happens at initialization and at window resize
```

Comment:
Since you below call `setPrefWidth`, after a window resize the `expansionShare` no longer reflect the actual preferred widths. 

I think the easiest solution is to use `setWidth` instead of `setPrefWidth` below: https://stackoverflow.com/a/53256244/873661

Reply:
We argue that the preferred size is the size the user set manually. It is difficult to distinguish whether a preferred size was set by JabRef automatically or by the user. During experiments with the functionality, it seems that maximization of the window and shrinkage and maximization and so on did not change the columns alignment. Thus, we opted to store the currently stored ratios.

For instance, that way when a user resizes the field "title" to twice the size of "author", this ratio is kept when resizing the window.

Reply:
Yeah, I was worried that you re-calculate the preferred width (e.g. in the initialization code), so this not necessarily need to correspond to the user's wish. But if that's not a problem, then nice.